docs/content/basics/testing_theory.md

```diff
@@ -1,57 +1,163 @@
 # Testing theory
 
 
-This article will cover the most primitive things that will allow you to understand better the terms and general concepts regarding testing.
+This article will cover the most primitive things that will allow you to understand better the terms and general concepts regarding automated testing.
 <br/>It is also worth noting that there will be an accent on UI tests, because we are in CookBook for UI tests.
 
-# Where it all starts
+
+## Where it all starts
 <i>Why do I need testing?</i> Users of your application/system/whatever should receive a quality product. 
 <br/>Thus, testing is about ensuring the quality of the product. Testing also includes creating a test plan, creating/conducting tests themselves and analyzing results of testing.
 
-Let's focus on the topic of creating tests. To begin with, let's see the testing pyramid.
+There are different means to classify tests. We will focus on the following
+
+1. *Granularity*
+2. *Running environment*
+3. *Target*
+
+## Classifying tests by granularity
+To begin with, let's see the testing pyramid.
 
 ![Testing pyramid](../images/pyramid.png "Testing pyramid")
 
+1. **End-to-End tests (E2E)**: these tests verify that all components in every single layer of the application work together as expected. They allow to test User stories - how our users are going to use the application. This makes them incredibly important both for business and for an ordinary developer or tester.
```

Comment:
This could be shorter. I believe it is better not to go very deep into details?
Or maybe the other way around?

Reply:
I guess in our case it's better to go deeper :)